packages/next-urql/src/types.ts

```diff
@@ -1,32 +1,45 @@
 import { GraphQLError } from 'graphql';
-import { NextPageContext } from 'next';
+import { ComponentType } from 'react';
 import { ClientOptions, Exchange, Client } from 'urql';
-import { AppContext } from 'next/app';
+
+export interface PartialNextContext {
+  res?: any;
+  AppTree?: NextComponentType<PartialNextContext>;
+  Component?: NextComponentType<PartialNextContext>;
+  ctx?: PartialNextContext;
```

Comment:
Is this to handle the case that `PartialNextContext` could be _either_ `NextPageContext` or `AppContext`? Looks fine to me, just wish there was a good way to limit recursive types in TS. I don't pretend to have enough knowledge of something like conditional types to make this cleaner, so looks good to me!

Reply:
I think we could also make this a union, but I thought since it's mostly used internally that won't be necessary. Instead the enhancer itself infers that the inout type is the output type 🤔